src/core/page_usage_stats_test.cc

```diff
@@ -176,3 +181,28 @@ TEST_F(PageUsageStatsTest, StatCollection) {
   const CollectedPageStats::ShardUsageSummary expected{{50, 65}, {90, 85}, {99, 89}};
   EXPECT_EQ(usage.at(1), expected);
 }
+
+TEST_F(PageUsageStatsTest, JSONCons) {
```

Comment:
Unfortunately it is not possible to test for flat encoding in CI due to the static lifetime of the encoding flag. 

Even if the test accesses internal fields of `json_obj` union directly to call `DefragIfNeeded`, the `Free` code path later on crashes trying to free a compact object which has an encoding which does not match the static setting.

I did test flat encoding manually in a matching test and it worked as expected.